#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_tcl8.3_testsuite_failure.dpatch by Jay
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix testsuite failure with tcl8.3

@DPATCH@
--- findutils-4.2.27/locate/testsuite/config/unix.exp	2005-06-08 00:24:56.000000000 +0200
+++ findutils/locate/testsuite/config/unix.exp	2005-11-26 08:54:45.000000000 +0100
@@ -27,10 +27,22 @@
 # We normalise (normalize for those over the water) pathnames 
 # because the updatedb shell script uses "cd", which means that 
 # any relative paths no longer point where we thought they did.
-set env(find) [file normalize "../../find/find"]
+# Because "file normalize" requires tcl 8.4, we have a plan B
+# for normalising the name of a directory, but it is slower.
+
+proc normalize_dir { dir } {
+    if [ catch { file normalize $dir } result ] then {
+	return [ exec /bin/sh -c "cd $dir && /bin/pwd" ]
+    } else {
+	return $result;
+    }
+}
+
+set fulldir [ normalize_dir "../../find" ]
+set env{find} "$fulldir/find"
 
 # use the local help commands for updatedb
-set env(LIBEXECDIR) [file normalize .. ]
+set env(LIBEXECDIR) [ normalize_dir .. ] 
 
 # do not ignore any file systems for this test
 set env(PRUNEFS) ""
